src/Spice86.Core/Emulator/OperatingSystem/DosPathResolver.cs

```diff
@@ -0,0 +1,315 @@
+namespace Spice86.Core.Emulator.OperatingSystem;
+
+using Spice86.Core.Emulator.OperatingSystem.Enums;
+using Spice86.Core.Emulator.OperatingSystem.Structures;
+using Spice86.Shared.Utils;
+
+using System.IO;
+using System.Linq;
+using System.Text;
+
+/// <summary>
+/// Translates DOS filepaths to host file paths, and vice-versa.
+/// </summary>
+internal class DosPathResolver {
+    internal const char VolumeSeparatorChar = ':';
+    internal const char DirectorySeparatorChar = '\\';
+    private const char AltDirectorySeparatorChar = '/';
+    private const int MaxPathLength = 255;
+
+    /// <summary>
+    /// Initializes a new instance.
+    /// </summary>
+    /// <param name="configuration">The emulator configuration.</param>
+    public DosPathResolver(Configuration configuration) {
+        _driveMap = InitializeDriveMap(configuration);
+        _currentDrive = 'C';
```

Comment:
Maybe move this to the parameters with a default of 'C', or add the default to the configuration?

Reply:
Done.